### PR TITLE
evremap: init at 0-unstable-2024-06-17, nixos/evremap: init module

### LINKED ISSUE #352471: Package request: evremap
**Project description**

A keyboard input remapper for Linux/Wayland systems

**Metadata**

* homepage URL: https://github.com/wez/evremap
* source URL: https://github.com/wez/evremap
* license: mit
* platforms: linux

Note: there has been an earlier abandoned PR (#295153)

### PR DESCRIPTION
**Pull request by pluiedev**
Fixes #352471

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by KiaraGrouwstra** [comment]
apologies for the late response. the package seems ok after resolving the conflict - i can post the result at discourse if it's resolved, sorry for the trouble.

**Comment by pluiedev** [comment]
Conflict resolved - ready to merge

**Comment by KiaraGrouwstra** [comment]
- [x] reviewed the diff and commit messages
- [x] made sure ofBorg build succeeded for all applicable platforms
- [x] run nix-review for a reasonable amount of time without any failures (or marked preexisting failures as broken)

File: nixos/modules/services/misc/evremap.nix [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        for how to configure evremap.
+      '';
+      default = { };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ pkgs.evremap ];
+
+    hardware.uinput.enable = lib.mkDefault true;
```
Comment (Scrumplex) [inline nixos/modules/services/misc/evremap.nix:114]: If this is required for this module to work, mkDefault isn't the right choice.

```suggestion
    hardware.uinput.enable = true;
```

File: nixos/modules/services/misc/evremap.nix [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+    systemd.services.evremap = {
+      description = "evremap service";
```
Comment (Scrumplex) [inline nixos/modules/services/misc/evremap.nix:117]: nit

```suggestion
      description = "evremap - keyboard input remapper";
```

**Comment by KiaraGrouwstra** [comment]
Result of `nixpkgs-review pr 352720` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>evremap</li>
  </ul>
</details>

**Review by Scrumplex — COMMENTED** [review]
Changes LGTM. I wasn't able to test this locally as the service crashed with the following output:

```
Nov 09 12:46:07 andromeda systemd[1]: Started evremap - keyboard input remapper.
Nov 09 12:46:07 andromeda evremap-start[264240]: [2024-11-09T11:46:07Z WARN  evremap] Short delay: release any keys now!
Nov 09 12:46:09 andromeda evremap-start[264240]: Error: creating UInputDevice from /dev/input/event0
Nov 09 12:46:09 andromeda evremap-start[264240]: Caused by:
Nov 09 12:46:09 andromeda evremap-start[264240]:     Permission denied (os error 13)
Nov 09 12:46:09 andromeda systemd[1]: evremap.service: Main process exited, code=exited, status=1/FAILURE
Nov 09 12:46:09 andromeda systemd[1]: evremap.service: Failed with result 'exit-code'.
lines 31-101/101 (END)
```

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 352720`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>evremap</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>evremap</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>evremap</li>
  </ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**